metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/jobs.py

```diff
@@ -255,14 +255,19 @@ class JobDirection(Enum):
     DOWNSTREAM = "downstream"
 
 
-@dataclass(frozen=True)
+@dataclass()
 class ModelJob:
     """
     Intermediate representation of a job's related models. Subsequently used by the SageMaker jobs ingestion framework.
     """
 
     job_urn: str
     job_direction: JobDirection
+    hyperparameters: Dict[str, Any] = field(default_factory=dict)
+    metrics: Dict[str, Any] = field(default_factory=dict)
+
+    def __hash__(self):
+        return hash((self.job_urn, self.job_direction))
```

Comment:
this seems a bit odd to me?

is there a reason we can't use `frozendict`? it's a hashable type so we can continue using `frozen=True`

Reply:
I considered that, but in the end we're only looking for unique jobs by direction and name – if two jobs have different metrics/hyperparams then they must be distinct anyway